tests/integration/kafka/test_consumer.py

```diff
@@ -25,7 +25,11 @@ class TestPartitionsForTopic:
     def test_partitions_for_returns_empty_for_unknown_topic(self, consumer: KafkaConsumer) -> None:
         assert consumer.partitions_for_topic("nonexistent") == {}
 
-    def test_partitions_for(self, consumer: KafkaConsumer, new_topic: NewTopic) -> None:
+    def test_partitions_for(
+        self,
+        new_topic: NewTopic,
+        consumer: KafkaConsumer,
+    ) -> None:
```

Comment:
its actually the order of the fixture that causes the problem?

Reply:
this is reaaaaally bad, I would expect this to be uninfluent. The order of parameters in a function shouldn't change the semantics of a program.
I wasn't quite liking the idea of fixture but this is much more dangerous.
Shouldn't we create a higher order fixture that gives the parameters and enforces the right order?

IMO if we need a topological sort of the fixture generation we should enforce it by definition